com/win32com/client/__init__.py

```diff
@@ -123,6 +123,10 @@ class CDispatch(dynamic.CDispatch):
   def _wrap_dispatch_(self, ob, userName = None, returnCLSID = None, UnicodeToString=None):
     assert UnicodeToString is None, "this is deprecated and will go away"
     return Dispatch(ob, userName, returnCLSID,None)
+  def __dir__(self):
+    import warnings
+    warnings.warn("__dir__() on auto-dynamic %r" % self)
```

Comment:
Why this warning? If it should be there, we should at least tell the user what they can do to avoid it or why it's being emitted - it would look to the user like a stray debugging print than an actionable warning.